src/inference/include/ie/ie_extension.h

```diff
@@ -19,10 +19,16 @@
 
 namespace InferenceEngine {
 
+IE_SUPPRESS_DEPRECATED_START
 /**
+ * @deprecated The Inference Engine API is deprecated and will be removed in the 2024.0 release. For instructions on
+ * transitioning to the new API, please refer to https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html
  * @brief This class is a C++ helper to work with objects created using extensions.
  */
-class INFERENCE_ENGINE_API_CLASS(Extension) final : public IExtension {
+class INFERENCE_ENGINE_DEPRECATED(
+    "The Inference Engine API is deprecated and will be removed in the 2024.0 release. For instructions on "
+    "transitioning to the new API, please refer to https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html")
```

Comment:
I suppose we can create a placeholder INFERENCE_ENGINE_1_0_DEPRECATED which will be expanded to extensive description message.